### PR TITLE
lsyncd: patch CVE-2014-8990

### PR DESCRIPTION
**Pull request by tdsmith**
Commit from master doesn't apply cleanly because of formatting changes.

### CODE REVIEW

**Comment by tdsmith** [comment]
on yosemite:
> Error: lsyncd does not define resource "xnu"

What is MACOS_FULL_VERSION on the yosemite bot?

**Comment by DomT4** [comment]
On 10.10.2 I'm getting:

```
Warning: No xnu headers for 10.10.2
==> Patching
/usr/bin/patch -g 0 -f -p1 -i /private/tmp/lsyncd--patch-N93XnR/cve-2014-8990.diff
patching file default-rsyncssh.lua
/usr/local/Library/Homebrew/debrew.rb:11:in `raise'
ResourceMissingError: lsyncd does not define resource "xnu"
```

**Comment by tdsmith** [comment]
How come I don't have 10.10.2? :crying_cat_face: I feel all left out.

Next question: how to choose the correct version of xnu headers from https://opensource.apple.com/source/xnu/ for 10.10.2?

**Comment by tdsmith** [comment]
Ah, it's reported by uname:

```
darkside:Developer tim$ uname -a
Darwin darkside.local 14.0.0 Darwin Kernel Version 14.0.0: Fri Sep 19 00:26:44 PDT 2014; root:xnu-2782.1.97~2/RELEASE_X86_64 x86_64
```

@DomT4, could you give me `uname -a` output for 10.10.2?

**Comment by DomT4** [comment]
> How come I don't have 10.10.2?

As far as I know, It's still on seed 3 as per today, not stable released yet. Maybe the Bot is hooked into a Beta program or something?

> could you give me uname -a output for 10.10.2?

Sure:

```
Darwin Dominyks-MacBook-Pro.local 14.1.0 Darwin Kernel Version 14.1.0: Thu Jan  8 00:18:18 PST 2015; root:xnu-2782.10.72~4/RELEASE_X86_64 x86_64
```

**Comment by tdsmith** [comment]
Hmm, wacky. Think I'll sit on this until 10.10.2 is final or the bot is reverted or something else happens. @mikemcquaid, is the pre-release update on the bot intentional?

**Comment by tdsmith** [comment]
(Thanks for the assist, Dominyk!)

**Comment by DomT4** [comment]
No worries! Feel free to shout if I can be useful for anything at all.

File: Library/Formula/lsyncd.rb [hunk 1]
```diff
@@ -48,6 +48,16 @@ class Lsyncd < Formula
       url "http://www.opensource.apple.com/tarballs/xnu/#{tarball}"
       sha1 checksum
     end
+  else
+    opoo "No xnu headers for #{MACOS_FULL_VERSION}"
+  end
+
+  # patch for CVE-2014-8990
+  # https://github.com/axkibe/lsyncd/commit/e9ffda07f0145f50f2756f8ee3fb0775b455122b
+  # https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2014-8990
```
Comment (mikemcquaid) [inline Library/Formula/lsyncd.rb]: :heart: for adding the URLs here.

**Comment by jacknagel** [comment]
We can just use the most recent available headers for 10.10.2, and then update again when Apple posts a newer version.

**Comment by DomT4** [comment]
10.10.2 is officially stable as of 12 hours ago or so. ` uname ` has changed very slightly, they actually seem to have rolled the number back a little bit for the stable release:

` Darwin Dominyks-MacBook-Pro.local 14.1.0 Darwin Kernel Version 14.1.0: Mon Dec 22 23:10:38 PST 2014; root:xnu-2782.10.72~2/RELEASE_X86_64 x86_64 ` 

Nothing matching that on the open-source website yet though.

**Pull request closed without merging**